Sources/web3swift/Utils/EIP/EIP681.swift

```diff
@@ -227,5 +347,202 @@ extension Web3 {
+
+        /// Attempts to split given ``rawValue`` into `[String]` where each element of that array
+        /// represent a valid, stringified, array in of itself.
+        /// With an input like `"[[],[],[]]"` the output is expected to be `["[]","[]","[]"]`.
+        /// - Parameter rawValue: supposedly an array of arrays in a form `[[...],[...],...]`
+        /// - Returns: separated stringified arrays, or `nil` if separation failed.
+        private static func splitArrayOfArrays(_ rawValue: String) -> [String]? {
+            /// Dropping first and last square brackets.
+            /// That modifies the upper bound value of the first match of `squareBracketRegex`.
+            let rawValue = String(rawValue.dropFirst().dropLast())
+
+            let squareBracketRegex = try! NSRegularExpression(pattern: "(\\[*)")
+            let match = squareBracketRegex.firstMatch(in: rawValue, range: rawValue.fullNSRange)
+
+            guard let bracketsCount = match?.range.upperBound,
+                  bracketsCount > 0 else {
+                return nil
+            }
+
+            let splitRegex = try! NSRegularExpression(pattern: "(\\]){\(bracketsCount)},(\\[){\(bracketsCount)}")
+            var indices: [Int] = splitRegex.matches(in: rawValue, range: rawValue.fullNSRange)
+                .map { $0.range.lowerBound + bracketsCount }
+            if !indices.isEmpty {
+                indices.append(rawValue.count)
+                var prevIndex = 0
+                var result = [String]()
+                for index in indices {
+                    result.append(rawValue[prevIndex..<index])
+                    prevIndex = index + 1
+                }
+                return result
+            }
+            return [rawValue]
+        }
+
+        /// Attempts to split a string that represents an array of strings.
+        /// Example:
+        ///
+        ///      // input
+        ///      "[\"1\",\"abcd,efgh\",\"-=-=-\"]"
+        ///      // output
+        ///      ["1","abcd,efgh","-=-=-"]
+        ///
+        ///      // input
+        ///      "[1,abcd,efgh,-=-=-]"
+        ///      // output
+        ///      ["1","abcd","efgh","-=-=-"]
+        ///
+        /// - Parameter rawValue: stringified array of strings.
+        /// - Returns: an array of separated individual elements, or `nil` if separation failed.
+        private static func splitArrayOfStrings(_ rawValue: String) -> [String]? {
+            /// Dropping first and last square brackets to exclude them from the first and the last separated element.
+            let rawValue = String(rawValue.dropFirst().dropLast())
+
+            let elementsBoundary = try! NSRegularExpression(pattern: "\",\"")
+            var indices = Array(elementsBoundary
+                .matches(in: rawValue, range: rawValue.fullNSRange)
+                .map { result in
+                    result.range.lowerBound + 1
+                })
```

Comment:
And the same question is here, the shield characters that you're handling here could be be stripped by `JSONDecoder` and then clean structure could be printed by `RawRepresentable`. I'm not sure how it fits into exact that case, but here's not yet perfect but workable `RawRepresentable` way to made clean string represent of an `array`.

Reply:
I'm resolving this as I've added a TODO.